src/__tests__/patients/visits/VisitForm.test.tsx

```diff
@@ -21,17 +18,20 @@ describe('Visit Form', () => {
   } as Visit
 
   const setup = (disabled = false, initializeVisit = true, error?: any) => {
-    onVisitChangeSpy = jest.fn()
-    const mockPatient = { id: '123' } as Patient
-    return render(
-      <VisitForm
-        patient={mockPatient}
-        onChange={onVisitChangeSpy}
-        visit={initializeVisit ? visit : {}}
-        disabled={disabled}
-        visitError={error}
-      />,
-    )
+    const onVisitChangeSpy = jest.fn()
+    const TestComponent = () => {
+      const [visit2, setVisit] = React.useState(initializeVisit ? visit : ({} as Visit))
+      onVisitChangeSpy.mockImplementation(setVisit)
+      return (
+        <VisitForm
+          onChange={onVisitChangeSpy}
+          visit={visit2}
+          disabled={disabled}
+          visitError={error}
+        />
+      )
+    }
```

Comment:
I'm not sure how I feel about having a test-only component...

Is there any chance that we can combine these VisitForm tests into the test suite that checks the component rendering this form? In fact, we could probably do this with a lot of the `*Form` tests... :thinking:

Reply:
we could definitely remove the test for editing if that is out of scope for this file, but if we `{enter}` in the input field, it will actually call `onChange` handler => we need a real onChange to modify the `visit` prop

Reply:
IMHO it should be done in separate issue/PR (unless the big test to edit whole Form already exits)

Reply:
@Aprillion ViewVisit and AddVisitModal both use this form. I can see why they were going for a "only test the form once" deal, but that doesn't give any confidence that both the ViewVisit and AddVisitModal perform as expected.

And nicely enough, between those two tests it checks all the edge cases - ViewVisit should be readonly (i.e. disabled) and AddVisitModal should be usable (i.e. tests the onChange etc.)

Reply:
hmm.. ok, file deleted.